gdraw/ggadgetP.h

```diff
@@ -375,7 +376,10 @@ typedef struct gtabset {
     unsigned int fill1line: 1;
     unsigned int vertical: 1;
     unsigned int nowindow: 1;
+    bool closable, movable;
     FontInstance *font;
+    void (*remove_sync)(GWindow gw, int pos);
```

Comment:
The indirection through these function pointers seems to follow the model of `nested_expose` and `nested_mouse` below, but those appear to be unused at this point. 

I take it the idea is that the gtabset is a general widget that currently happens to be used only in CharViews, and this indirection preserves that modularity.

Reply:
Before I begin, I should correct you on one fact: the GTabSet GDraw widget actually finds itself in use in other places, notably the MM dialog. Grep for `GTabSetCreate`.

Adding on to my comment below, the reason we need these pointers is because the CharView is necessarily the most complicated case UI wise for the GTabSet. 

The CharView has its own tab metadata, which it stores in `cvtabs` (type `CharViewTab`). There's also `former_names` (metadata of a kind) and `oldnum`.

We have to keep this stuff updated or it will break in subtle ways. The reason this took two days and not two hours was this fundamental problem: keeping the GTabSet's metadata (`struct tabs*`) up to date with the CharView's metadata (`CharViewTab*`).

I do follow the model of the `nested` functions yes but even without them as a guide we need some way to keep both sets of metadata in sync.